tests/integration/test_failed_mutations/test.py

```diff
@@ -0,0 +1,169 @@
+    )
+
+    assert node.contains_in_log(POSPONE_MUTATION_LOG) == found_in_log
+    node.rotate_logs()
+
+    time.sleep(5)
+    node.query("KILL MUTATION WHERE table='test_mutations'")
+    # Check that after kill new parts mutations are postponing.
+    node.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM notexist_table) SETTINGS allow_nondeterministic_mutations=1"
+    )
+
+    assert node.contains_in_log(POSPONE_MUTATION_LOG) == found_in_log
+
+
+def test_exponential_backoff_with_replicated_tree(started_cluster):
+    prepare_cluster(True)
+
+    node_with_backoff.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM notexist_table) SETTINGS allow_nondeterministic_mutations=1"
+    )
+
+    assert node_with_backoff.wait_for_log_line(REPLICATED_POSPONE_MUTATION_LOG)
+    assert not node_no_backoff.contains_in_log(REPLICATED_POSPONE_MUTATION_LOG)
+
+
+@pytest.mark.parametrize(
+    ("node"),
+    [
+        (node_with_backoff),
+    ],
+)
+def test_exponential_backoff_create_dependent_table(started_cluster, node):
+    prepare_cluster(False)
+
+    # Executing incorrect mutation.
+    node.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM dep_table) SETTINGS allow_nondeterministic_mutations=1"
+    )
+    time.sleep(5)
+    # Creating dependent table for mutation.
+    node.query("CREATE TABLE dep_table(x UInt32) ENGINE MergeTree() ORDER BY x")
+
+    time.sleep(5)
+    assert node.query("SELECT count() FROM system.mutations WHERE is_done=0") == "0\n"
+    node.query("DROP TABLE IF EXISTS dep_table SYNC")
+
+
+def test_exponential_backoff_setting_override(started_cluster):
+    node = node_with_backoff
+    node.rotate_logs()
+    node.query("DROP TABLE IF EXISTS test_mutations SYNC")
+    node.query(
+        "CREATE TABLE test_mutations(x UInt32) ENGINE=MergeTree() ORDER BY x SETTINGS max_postpone_time_for_failed_mutations=0"
+    )
+    node.query("INSERT INTO test_mutations SELECT * FROM system.numbers LIMIT 10")
+
+    # Executing incorrect mutation.
+    node.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM dep_table) SETTINGS allow_nondeterministic_mutations=1"
+    )
+    assert not node.contains_in_log(POSPONE_MUTATION_LOG)
+
+
+@pytest.mark.parametrize(
+    ("replicated_table"),
+    [
+        (False),
+        (True),
+    ],
+)
+def test_backoff_clickhouse_restart(started_cluster, replicated_table):
+    prepare_cluster(replicated_table)
+    node = node_with_backoff
+
+    # Executing incorrect mutation.
+    node.query(
+        "ALTER TABLE test_mutations DELETE WHERE x IN (SELECT x  FROM dep_table) SETTINGS allow_nondeterministic_mutations=1"
+    )
+    assert node.wait_for_log_line(
+        REPLICATED_POSPONE_MUTATION_LOG if replicated_table else POSPONE_MUTATION_LOG
+    )
+
+    node.restart_clickhouse()
```

Comment:
Please use functional tests whenever possible: https://github.com/ClickHouse/ClickHouse/pull/39359#issuecomment-1188296170

In this case, `restart_clickhouse` can be replaced with DETACH/ATTACH TABLE, and `wait_for_log_line` with reading from system.text_log.

It's just FYI, it's not necessary to rewrite the test.